agent/eni/iphelperwrapper/iphelper_windows.go

```diff
@@ -0,0 +1,163 @@
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package iphelperwrapper
+
+import (
+	"C"
+	"math"
+	"syscall"
+	"time"
+	"unsafe"
+
+	log "github.com/cihub/seelog"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+const (
+	// After registering a callback with NotifyIPInterfaceChange API, we get an initial notification to confirm that the callback is registered
+	// All other notifications sent on the channel would be positive except this one
+	InitialNotification = math.MinInt16
+	// This is the timeout for the initial notification. If notification is not received within this timeout then we will assume error
+	InitialNotificationTimeout = 3 * time.Second
+	// This is the delay following the receipt of Initial Notification, after which the corresponding message is sent on the channel
+	// This has been done to ensure that the control reaches 'initialNotificationLoop' before the initial notification is sent
+	// Also, note that the initial notification will only be sent once. Therefore, this delay would be for the first communication only
+	DelayInCallbackMessageForInitialNotification = 10 * time.Millisecond
+)
+
+var (
+	moduleiphlpapi = windows.NewLazySystemDLL("iphlpapi.dll")
+
+	procNotifyIPInterfaceChange = moduleiphlpapi.NewProc("NotifyIpInterfaceChange")
+	proccancelMibChangeNotify2  = moduleiphlpapi.NewProc("CancelMibChangeNotify2")
+
+	// Added these to allow unit testing for this module. We will swap these System calls with dummy calls in unit tests
+	funcNotifyIPInterfaceChange = procNotifyIPInterfaceChange.Call
+	funcCancelMibChangeNotify2  = proccancelMibChangeNotify2.Call
+
+	// The notification channel is used for communication between the notification callback from Windows APIs and the consumer
+	notificationChannel chan int
+)
+
+// Interface for using the monitor
+// We will mock this interface while unit testing other packages
+type InterfaceMonitor interface {
+	StartMonitor(notification chan int) error
+	Close() error
+}
+
+type monitoringInterface struct {
+	notificationHandle windows.Handle
+}
+
+// Returns a new monitoringInterface which conforms to InterfaceMonitor interface
+func NewMonitor() (monitor *monitoringInterface) {
+	monitor = &monitoringInterface{}
+
+	notificationHandle := windows.Handle(0)
+	monitor.notificationHandle = notificationHandle
+
+	return
+}
+
+// This method is used to start the monitoring of new interfaces attached to the instance.
+// We register a notification callback with Windows API for any changes in instance interfaces (NotifyIPInterfaceChange).
+// Initial notification received within timeout is considered as a confirmation of successful registration
+// notificationChannel is used for any communication between the callback and the consumer
+func (monitor *monitoringInterface) StartMonitor(notification chan int) error {
```

Comment:
`monitor.StartMonitor` does not read very well.  This can simply be `Start`

Reply:
Implemented this change.